File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/lang/ClientLanguage.java
package pize.tests.minecraft.game.lang;

import pize.files.Resource;
import pize.files.MapConfig;

import java.util.HashMap;

public class ClientLanguage{

    private HashMap<String, String> storage;

    public static ClientLanguage loadFrom(Resource res){
        ClientLanguage language = new ClientLanguage();
        MapConfig map = new MapConfig(res);
        map.load();
        language.storage = map.map;

        return language;
    }

    public String getOrDefault(String key){
        return storage.getOrDefault(key, key);
    }

}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/lang/ClientLanguage.kt
package pize.tests.minecraft.game.lang

import pize.files.MapConfig
import pize.files.Resource

class ClientLanguage {
    private var storage: HashMap<String?, String?>? = null
    fun getOrDefault(key: String?): String? {
        return storage!!.getOrDefault(key, key)
    }

    companion object {
        fun loadFrom(res: Resource?): ClientLanguage {
            val language = ClientLanguage()
            val map = MapConfig(res)
            map.load()
            language.storage = map.map
            return language
        }
    }
}
